### PR TITLE
fix to empirical distribution block bugs

### PR DESCRIPTION
**Pull request by astrolamb**

### CODE REVIEW

**Review by paulthebaker — COMMENTED** [review]
The tests are passing, but they don't touch all of the added code.  It looks like there's a duplication in the try-except statement that I don't understand.  I just want to double check that it works as you intend.

File: enterprise_extensions/sampler.py [hunk 1]
```diff
@@ -46,8 +47,35 @@ def __init__(self, pta, snames=None, empirical_distr=None, f_stat_file=None):
...
+        
+            dir_files = glob.glob(empirical_distr+'*.pkl') # search for pkls
+        
+            pickled_distr = np.array([])
+            for idx, emp_file in enumerate(dir_files):
+                try:
+                    with open(emp_file, 'rb') as f:
+                        pickled_distr = np.append(pickled_distr, pickle.load(f))
+                except:
+                    try:
```
[the remark this answers is not in the document]
Comment (astrolamb) [reply enterprise_extensions/sampler.py:66]: There is no difference - this was something that it was in the previous version of the code.
I assume the author of the last version wanted to try to open the file again if the first attempt failed. Sometimes I've noticed that with pickle - if it fails the first time, you just need to try again.
I'll add a comment about this

[the remark this answers is not in the document]
Comment (astrolamb) [reply enterprise_extensions/sampler.py:66]: after adding the comments on my fork, do I just open another PR?

File: enterprise_extensions/sampler.py [hunk 2]
```diff
@@ -46,8 +47,35 @@ def __init__(self, pta, snames=None, empirical_distr=None, f_stat_file=None):
...
+        elif empirical_distr is not None and os.path.isdir(empirical_distr):
+        
+            dir_files = glob.glob(empirical_distr+'*.pkl') # search for pkls
+        
+            pickled_distr = np.array([])
+            for idx, emp_file in enumerate(dir_files):
+                try:
+                    with open(emp_file, 'rb') as f:
+                        pickled_distr = np.append(pickled_distr, pickle.load(f))
+                except:
```
[the remark this answers is not in the document]
Comment (astrolamb) [reply enterprise_extensions/sampler.py:65]: yeah it's catching everything and setting the problem argument to `None`, and printing a message to the output file saying that the intended input empirical files aren't added.

This is what the previous version of the code did. If you think that it's more appropriate, I can instead just throw an `FileNotFound` error and terminate the program?

[the remark this answers is not in the document]
Comment (paulthebaker) [reply enterprise_extensions/sampler.py:65]: If you don't want it to terminate, then don't.  I'm clarifying the intended behavior.  If you want it to print the warning and continue on a `FileNoFoundError`, that's what the except statement is for.

I'm saying that if that block throws some unexpected error, like a `TypeError`, maybe you don't want it continue on.

File: enterprise_extensions/sampler.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (paulthebaker) [reply enterprise_extensions/sampler.py:66]: When you commit them to the same branch the PR updates with the new changes

File: enterprise_extensions/sampler.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (astrolamb) [reply enterprise_extensions/sampler.py:65]: Sorry for not replying, I was preparing my talk for today!
I see that you've resolved this. I'll think about what to do about throwing a `TypeError` and send another PR if needed.
Thanks!

**Pull request merged**